testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -183,65 +184,74 @@ def _run(self, **kwargs):
             h.run(verbose=False)
         return h
 
-    def test_check_static_selections(self):
+    def run_HBA(self, s1, s2, s1type):
+        """test that HydrogenBondAnalysis() raises SelectionError for missing donors/acceptors"""
+        # no donors/acceptors; only raises error if no updates
+        return self._run(
+            selection1=s1,
+            selection2=s2,
+            update_selection1=False,
+            update_selection2=False,
+            selection1_type=s1type,
+        )
+
+    @pytest.mark.parametrize('s1, s2, s1type', itertools.product(
+        ("protein", "resname ALA and not backbone"),
+        ("protein", "resname ALA and not backbone"),
+        ("donor", "acceptor", "both")
+    ))
+    def test_check_static_selections_pt(self, s1, s2, s1type):
         self._setUp()
-        try:
-            def run_HBA(s1, s2, s1type):
-                """test that HydrogenBondAnalysis() raises SelectionError for missing donors/acceptors"""
-                # no donors/acceptors; only raises error if no updates
-                return self._run(selection1=s1, selection2=s2,
-                                 update_selection1=False, update_selection2=False,
-                                 selection1_type=s1type,
-                )
-            protein = "protein"
-            nothing = "resname ALA and not backbone"
-            for s1, s2, s1type in itertools.product((protein, nothing),
-                                                    (protein, nothing),
-                                                    ("donor", "acceptor", "both")):
-                if s1 == s2 == protein:
-                    def runOK():
-                        """test that HydrogenBondAnalysis() works for protein/protein"""
-                        try:
-                            h = run_HBA(s1, s2, s1type)
-                        except:
-                            raise AssertionError("HydrogenBondAnalysis protein/protein failed")
-                        else:
-                            return True
-                    yield runOK
+        protein = "protein"
+
+        if s1 == s2 == protein:
+            def runOK():
```

Comment:
The way the test is written now, what is the point of having the nested function?